File: solutions/198.HouseRobber/dj.cpp
#include <vector>
using namespace std;
class Solution {
public:
    int rob(vector<int>& nums) {
        vector<int> robDP(nums.size());
        vector<int> notrobDP(nums.size());
        robDP[0] = nums[0];
        for (int i = 1; i < nums.size(); ++i) {
            robDP[i] = notrobDP[i - 1] + nums[i];
            notrobDP[i] = max(robDP[i - 1], notrobDP[i - 1]);
        }

        return max(robDP[nums.size() - 1], notrobDP[nums.size() - 1]);
    }
};

File: solutions/740.DeleteandEarn/dj.py
class Solution:
    def deleteAndEarn(self, nums: List[int]) -> int:
        cnt = collections.Counter(nums)
        arr = [0] * (10**4 + 1)
        for k in cnt:
            arr[k] = cnt[k] * k
        
        dp = [0] * (10**4 + 1)
        dp[0] = arr[0]
        dp[1] = max(arr[0], arr[1])
        for i in range(10**4 + 1):
            dp[i] = max(dp[i - 1], dp[i - 2] + arr[i])
        
        return dp[-1]